packages/firestore/src/local/memory_query_cache.ts

```diff
@@ -65,15 +67,29 @@ export class MemoryQueryCache implements QueryCache {
     return PersistencePromise.resolve();
   }
 
-  addQueryData(
-    transaction: PersistenceTransaction,
-    queryData: QueryData
-  ): PersistencePromise<void> {
+  private saveQueryData(queryData: QueryData) {
     this.queries.set(queryData.query, queryData);
     const targetId = queryData.targetId;
     if (targetId > this.highestTargetId) {
       this.highestTargetId = targetId;
     }
+    // TODO(gsoltis): track sequence number
+  }
+
+  addQueryData(
+    transaction: PersistenceTransaction,
+    queryData: QueryData
+  ): PersistencePromise<void> {
```

Comment:
Can we assert that it already exists?  That a) protects us from bugs, and b) makes this code more obviously correct [the reader doesn't have to wonder if the targetCount accounting needs to take into account prior existence].

Reply:
Asserts added to memory version.